### ISSUE TITLE
IPython cuts string with '# coding: utf-8' text

### ISSUE THREAD

**Issue by gtors** [ipython/ipython#9919]
```
Python 3.5.2 (default, Jun 28 2016, 08:46:01)
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: s = '''#!/usr/bin/python
   ...: # coding: utf-8
   ...: '''

In [2]: s
Out[2]: '#!/usr/bin/python\n\n'

In [3]:
```

As you can see, Out[2] not contains `# coding: utf-8` ([asciinema](https://asciinema.org/a/epx1zqn6af4oj9kh4befsuf9b))

**Comment by takluyver** [comment]
We do this because if you paste in a file on Python 2, we read it as unicode, and when we ask Python to parse it, we get an error:

```
SyntaxError: encoding declaration in Unicode string
```

However, it seems that Python 3 ignores the encoding declaration instead of throwing an error, which makes our lives easier. So this can be fixed in 6.0 by simply not stripping the encoding declaration.

**Comment by takluyver** [comment]
#9924 should fix it.
